Dockerfile

```diff
@@ -1,30 +1,104 @@
-ARG ATLANTIS_BASE=ghcr.io/runatlantis/atlantis-base
-ARG ATLANTIS_BASE_TAG_DATE=latest
-ARG ATLANTIS_BASE_TAG_TYPE=alpine
+# syntax=docker/dockerfile:1
+# what distro is the image being built for
+ARG ALPINE_TAG=3.17.0
+ARG DEBIAN_TAG=bullseye-20221219-slim
 
-# Stage 1: build artifact
-
-FROM golang:1.19.5-alpine AS builder
+# Stage 1: build artifact and download deps
+FROM golang:1.19.4-alpine AS builder
 
 WORKDIR /app
 COPY . /app
 RUN --mount=type=cache,target=/go/pkg/mod \
     --mount=type=cache,target=/root/.cache/go-build \
     CGO_ENABLED=0 go build -trimpath -ldflags "-s -w" -v -o atlantis .
 
-# Stage 2
-# The runatlantis/atlantis-base is created by docker-base/Dockerfile
-FROM ${ATLANTIS_BASE}:${ATLANTIS_BASE_TAG_DATE}-${ATLANTIS_BASE_TAG_TYPE} AS base
+FROM debian:${DEBIAN_TAG} as deps
 
 # Get the architecture the image is being built for
 ARG TARGETPLATFORM
 
+# Install packages needed for building/verifying dependencies
+# hadolint ignore=DL3008,SC2261
+WORKDIR /tmp/build
+RUN apt-get update \
+    && apt-get install -y --no-install-recommends \
+        ca-certificates>=20210119 \
+        curl>=7.74 \
+        git>=1:2.30 \
+        unzip>=6.0 \
+        bash>=5.1 \
+        openssh-server>=1:8.4p1 \
+        libcap2>=1:2.44 \
+        dumb-init>=1.2 \
+        gnupg>=2.2 \
+        openssl>=1.1.1n
+
+# install conftest
+# renovate: datasource=github-releases depName=open-policy-agent/conftest
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
+ENV DEFAULT_CONFTEST_VERSION=0.38.0
+RUN AVAILABLE_CONFTEST_VERSIONS="${DEFAULT_CONFTEST_VERSION}" && \
+    case ${TARGETPLATFORM} in \
+        "linux/amd64") CONFTEST_ARCH=x86_64 ;; \
+        "linux/arm64") CONFTEST_ARCH=arm64 ;; \
+        # There is currently no compiled version of conftest for armv7
+        "linux/arm/v7") CONFTEST_ARCH=x86_64 ;; \
+    esac && \
+    for VERSION in ${AVAILABLE_CONFTEST_VERSIONS}; do \
+        curl -LOs https://github.com/open-policy-agent/conftest/releases/download/v${VERSION}/conftest_${VERSION}_Linux_${CONFTEST_ARCH}.tar.gz && \
+        curl -LOs https://github.com/open-policy-agent/conftest/releases/download/v${VERSION}/checksums.txt && \
+        sed -n "/conftest_${VERSION}_Linux_${CONFTEST_ARCH}.tar.gz/p" checksums.txt | sha256sum -c && \
+        mkdir -p /usr/local/bin/cft/versions/${VERSION} && \
+        tar -C /usr/local/bin/cft/versions/${VERSION} -xzf conftest_${VERSION}_Linux_${CONFTEST_ARCH}.tar.gz && \
+        ln -s /usr/local/bin/cft/versions/${VERSION}/conftest /usr/local/bin/conftest && \
+        rm conftest_${VERSION}_Linux_${CONFTEST_ARCH}.tar.gz && \
+        rm checksums.txt; \
+    done
+
+# install gosu
+# We use gosu to step down from root and run as the atlantis user
+# renovate: datasource=github-releases depName=tianon/gosu
+ENV GOSU_VERSION=1.16
```

Comment:
I haven't seen any flakiness with the test builds, however, if there is an issue down the line we can address the issue differently rather than separating the dockerfiles. The base changes now that we no longer pin packages on =.

Reply:
One problem we will run into is that if the "base" portion of the image fails due to packages not being found, it will come up on a PR which will force the user to have to resolve it, rather than it getting resolved outside.

This was the main reason it was split up so that way the atlantis-image could always succeed unless the PR specific changes caused it to fail

Reply:
Ref https://github.com/runatlantis/atlantis/pull/179

cc: @lkysow

Reply:
@nitrocode Understandable. Didn't realize alpine removed packages in such a fashion. Shouldn't we be more resilient to this then since we don't pin as aggressively since https://github.com/runatlantis/atlantis/pull/2896? 

I think the benefits here (less complexity) outweigh the risk (failed build). We can easily update packages outside of the PR and have the author rebase.